tests/test_filters.py

```diff
@@ -172,3 +173,12 @@ def test_row_col_medians_with_mask(
     assert isinstance(medians["cols"], np.ndarray)
     np.testing.assert_array_equal(medians["rows"], image_random_row_medians_masked)
     np.testing.assert_array_equal(medians["cols"], image_random_col_medians_masked)
+
+
+def test_non_square_img(test_filters_random: Filters):
+    test_filters_random.images["pixels"]=test_filters_random.images["pixels"][:,0:512]
+    test_filters_random.images["zero_averaged_background"] = test_filters_random.average_background(
+        test_filters_random.images["pixels"], test_filters_random.images["pixels"]
```

Comment:
Been a while since I've looked at this but why do you pass the same image in as a mask here?

The `average_background()` method passes it to `row_col_medians()` where it is conditionally used to mask values using `np.ma.masked_array()` and the `mask` argument to this is expected to be Boolean. Since most values are non-zero the whole image gets masked.

In turn this means that we can't test the values stored in `test_filters_random.images["zero_averaged_background"]` and test what they are rather than just whether its a numpy array and has the desired shape (see below comment).

I'm not sure if I've articulated this well so have tried to create a Minimal Reproducible Example based on the code in `Filters()` class to show what is happening...

```python
import numpy as np
square_array = np.array([[10,11,12,13,14],
                         [24,26,29,40,21],
                         [39,89,43,11,10],
                         [56,57,20,95,11],
                         [15,89,35,46,45]])
non_square_array = square_array[0:3,]

def row_medians(x):
    return np.nanmedian(x, axis=1)

def average_background(x, mask=False):
    if mask is False:
        return (x.T - row_medians(x)).T
    else:
        masked = np.ma.masked_array(x, mask=x)
        return (masked.T - row_medians(masked)).T

# Square and no mask
average_background(square_array, mask=False)
array([[ -2.,  -1.,   0.,   1.,   2.],
       [ -2.,   0.,   3.,  14.,  -5.],
       [  0.,  50.,   4., -28., -29.],
       [  0.,   1., -36.,  39., -45.],
       [-30.,  44., -10.,   1.,   0.]])
# This is ok, its what we expect.

# Square with mask
average_background(square_array, mask=True)
masked_array(
  data=[[--, --, --, --, --],
        [--, --, --, --, --],
        [--, --, --, --, --],
        [--, --, --, --, --],
        [--, --, --, --, --]],
  mask=[[ True,  True,  True,  True,  True],
        [ True,  True,  True,  True,  True],
        [ True,  True,  True,  True,  True],
        [ True,  True,  True,  True,  True],
        [ True,  True,  True,  True,  True]],
  fill_value=1e+20,
  dtype=float64)
# A masked array isn't what we want returned, in Filters() masked values are filled with np.nan and so it would sum() to nan

# Non-square no mask
average_background(non_square_array, mask=False)
array([[ -2.,  -1.,   0.,   1.,   2.],
       [ -2.,   0.,   3.,  14.,  -5.],
       [  0.,  50.,   4., -28., -29.]])
# Again good, this is what we want

# Non-square with mask
average_background(non_square_array, mask=True)
masked_array(
  data=[[--, --, --, --, --],
        [--, --, --, --, --],
        [--, --, --, --, --]],
  mask=[[ True,  True,  True,  True,  True],
        [ True,  True,  True,  True,  True],
        [ True,  True,  True,  True,  True]],
  fill_value=1e+20,
  dtype=float64)
# Not good, we've masked everything and again in Filters() masked values are filled with np.nan and so it would sum() to nan
```

This is a test scenario so I guess one question might be "_Why does this matter, when using we won't ever use the image itself to mask, we use the derived mask_" which is true, but as I've asked below, I think we should be testing the `.sum()` of the returned array to ensure that we always get the same answer rather than just an array with the expected size.

Reply:
Test resolved via mask=None